js/exporter/jspdf/export.js

```diff
@@ -0,0 +1,181 @@
+
+        const tableWidth = component.option('width');
+        const dataProvider = component.getDataProvider(selectedRowsOnly);
+
+        return new Promise((resolve) => {
+            dataProvider.ready().done(() => {
+                const columns = dataProvider.getColumns();
+                const styles = dataProvider.getStyles();
+                const dataRowsCount = dataProvider.getRowsCount();
+
+                this.setTableWidth(autoTableOptions, tableWidth);
+                this.setColumnWidths(autoTableOptions, columns, keepColumnWidths);
+
+                for(let rowIndex = 0; rowIndex < dataRowsCount; rowIndex++) {
+
+                    let rowType = null;
+                    const row = [];
+
+                    for(let cellIndex = 0; cellIndex < columns.length; cellIndex++) {
+
+                        const cellData = dataProvider.getCellData(rowIndex, cellIndex, true);
+                        const cellStyle = styles[dataProvider.getStyleId(rowIndex, cellIndex)];
+                        const gridCell = cellData.cellSourceData;
+                        const pdfCell = {
+                            content: cellData.value,
+                            styles: {}
+                        };
+
+                        this.assignCellStyle(pdfCell, gridCell, columns[cellIndex], cellStyle);
+
+                        rowType = rowType || gridCell.rowType;
+                        if(rowType === 'header') {
+                            if(pdfCell.content !== '') { row.push(pdfCell); }
+                        } else {
+                            row.push(pdfCell);
+                        }
+                    }
+
+                    this.addRow(autoTableOptions, rowType, row);
+                }
+
+                jsPDFDocument.autoTable(autoTableOptions);
+
+                resolve(autoTableOptions);
+            }).always(() => {
+
+            });
+        });
+    },
+
+    addRow: function(autoTableOptions, rowType, row) {
+        if(rowType === 'header') { autoTableOptions.head.push(row); }
+        if(rowType === 'data') { autoTableOptions.body.push(row); }
+    },
+
+    assignCellStyle: function(pdfCell, gridCell, column, cellStyle) {
+        if(gridCell.rowType === 'header') {
+            // eslint-disable-next-line spellcheck/spell-checker
+            if(column.alignment) { pdfCell.styles.halign = column.alignment; }
+        } else {
+            if(cellStyle) {
+                // eslint-disable-next-line spellcheck/spell-checker
+                if(cellStyle.alignment) { pdfCell.styles.halign = cellStyle.alignment; }
+                if(cellStyle.bold) { pdfCell.styles.fontStyle = 'bold'; }
+                if(cellStyle.wrapText) { pdfCell.styles.cellWidth = 'wrap'; }
+                if(cellStyle.dataType === 'date') { pdfCell.content = dateLocalization.format(new Date(pdfCell.content), cellStyle.format); }
+            }
+        }
+    },
+
+    setTableWidth: function(autoTableOptions, width) {
```

Comment:
Again 'one line of code' function.
Why don's use this line instead of setTableWidth call?